draft-ietf-jsonpath-base.md

```diff
@@ -1195,19 +1188,159 @@ compared is an object, array, boolean, or `null`.
+nodelist MUST return a singular node to be used in a comparable.
+
+###### Function Extensions: length {#length}
 {: unnumbered}
 
-The logical AND, OR, and NOT operators have the normal semantics of Boolean algebra and
-obey its laws (see, for example, {{BOOLEAN-LAWS}}).
+Arguments:
+: 1. value
+
+Result:
+: value (unsigned integer)
+
+The "length" function extension provides a way to compute the length
+of a value and make that available for further processing in the
+filter expression:
+
+~~~ JSONPath
+$[?length(@.authors) >= 5]
+~~~
+
+Its only argument is a value (possibly taken from a singular path as
+in the example above).  The result also is a value, an unsigned
+integer.
+
+* If the argument value is a string, the result is the number of
+  Unicode scalar values in the string.
+* If the argument value is an array, the result is the number of
+  elements in the array.
+* If the argument value is an object, the result is the number of
+  members in the object.
+* For any other argument value, the result is one.
+
+
+###### Function Extensions: count {#count}
+{: unnumbered}
+
+Arguments:
+: 1. nodelist
+
+Result:
+: value (unsigned integer)
+
+The "count" function extension provides a way to obtain the number of
+nodes in a nodelist and make that available for further processing in
+the filter expression:
+
+~~~ JSONPath
+$[?count(@.*.author) >= 5]
+~~~
+
+Its only argument is a nodelist.
+The result is a value, an unsigned integer, that gives the number of
+nodes in the nodelist.
+Note that there is no deduplication of the nodelist. [^dedup]
+
+[^dedup]: Well, that can be discussed.
+
+
+###### Function Extensions: match {#match}
+{: unnumbered}
+
+Arguments:
+: 1. value (string)
+  2. value (string conforming to {{-iregexp}})
+
+Result:
+: value (`true` or `false`)
+
+The "match" function extension provides a way to check whether a given
+string matches a given regular expression, which is in {{-iregexp}} form.
+
+~~~ JSONPath
+$[?match(@.date, "1974-05-..")]
+~~~
+
+Its first argument is a string that is matched against the iregexp
+contained in the string that is the second argument.
+The result is `true` if the string matches the iregexp and `false`
+otherwise.
+
+
+###### Function Extensions: search {#search}
+{: unnumbered}
+
+Arguments:
+: 1. value (string)
+  2. value (string conforming to {{-iregexp}})
+
+Result:
+: value (`true` or `false`)
+
+The "search" function extension provides a way to check whether a
+given string contains a substring that matches a given regular
+expression, which is in {{-iregexp}} form.
+
+~~~ JSONPath
+$[?search(@.author, "[BR]ob")]
+~~~
+
+Its first argument is a string that is searched for at least one
+substring that matches against the iregexp contained in the string
+that is the second argument.
+The result is `true` if such a substring exists, `false` otherwise.
+
+o o o o o o o o o o o o o o o o o o o o o o o o o o o o o o o o
 
 #### Examples
```

Comment:
Need examples of `match()` and `search()` to better highlight the difference between them.  The text varies only by one saying "contains a match" vs just "matches," which I don't think is clear.  In my experience with regexes, containing a match _is_ matching.

Reply:
Right.  The one example that has a match() would work the same with a search()... we probably need to set up a pair of contrasting examples.  Let's do a round through the examples once the technical content has stabilized.

Reply:
The text is fine though.  87d95d6 adds a few words that might help grasping the difference.